syntax="proto3";

import "google/protobuf/timestamp.proto";

message Worker{
        string worker_name=1;
        int32 worker_id=2;

        message Query{
        		bool result_found=1;
                string query=2;
                int32 query_id=3;
                string path=4;
        }

        message Index{
                bool indexed=1;
                bool file_added=2;
                bool file_removed=3;
                string file_name=4;
                string file_path=5;
        }

        repeated Query queries=3;
        repeated Index Indexes=4;

        google.protobuf.Timestamp last_indexed=5;
}

message WorkersLog{
        repeated Worker workers=1;
}
